examples/vault-dynamodb-backend/main.tf

```diff
@@ -0,0 +1,89 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# REQUIRE A SPECIFIC TERRAFORM VERSION OR HIGHER
+# This module has been updated with 0.12 syntax, which means it is no longer compatible with any versions below 0.12.
+# ----------------------------------------------------------------------------------------------------------------------
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY THE VAULT SERVER CLUSTER
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "aws_dynamodb_table" "vault_dynamo" {
```

Comment:
Unlike the original implementation, I left the DB out of the cluster module. 
https://github.com/hashicorp/terraform-aws-vault/compare/master...stephencoe:dynamodb_backend?expand=1#diff-31d3824adbff3c2b113b417aae8ee176R13

Doing this means you can use a global table if required rather than a single table. The downside of this is the need to pass the table arn into the cluster module for permissions since there is no data provider for a global table

https://github.com/hashicorp/terraform-aws-vault/compare/master...stephencoe:dynamodb_backend?expand=1#diff-31d3824adbff3c2b113b417aae8ee176R59